server/core/src/https/views/profile.rs

```diff
@@ -48,25 +75,140 @@ pub(crate) async fn view_profile_get(
 
     let profile_partial_view = ProfilePartialView {
         can_rw,
-        account_name: uat.name().to_string(),
-        display_name: uat.displayname.clone(),
-        legal_name: uat.name().to_string(),
-        email: uat.mail_primary.clone(),
+        attrs: ProfileAttributes {
+            account_name: uat.name().to_string(),
```

Comment:
At this point we could probably do a call to get the entry, then you can use the https://github.com/kanidm/kanidm/blob/master/server/lib/src/entry.rs#L2605 apis here to get the exact type and value you need. account_name is "iname", displayName and legalname is utf8, and for mail we might need to return something structured for you so you can handle the primary / not primary case.